src/routes/RequestsRoute.js

```diff
@@ -1171,22 +1184,43 @@ class RequestsRoute extends React.Component {
     );
   };
 
-  onBeforeGetContentForPrintButton = (onToggle) => (
-    new Promise(resolve => {
+  filterSelectedPickSlipsData = (pickSlipsData, selectedRow) => {
+    const selectedRequestIds = new Set(Object.keys(selectedRow));
+    return pickSlipsData
+      .filter(entry => selectedRequestIds.has(entry.request?.requestID));
+  }
+
+  savePrintEventDetails = (requestIds) => {
+    const isoDate = moment.tz('UTC').format();
+    this.props.mutator.savePrintDetails.POST({
+      'requestIds' : requestIds,
+      'requesterName' : this.props.stripes.user.user.username,
+      'requesterId' : this.props.stripes.user.user.id,
+      'printEventDate' : isoDate
+    });
+  }
+
+  onBeforeGetContentForPrintButton = (onToggle, slipsType, slipsData) => {
+    if (slipsType === SLIPS_TYPE.PICK_SLIP) { // add enable print log settings check
+      const requestIds = extractPickSlipRequestIds(slipsData);
+      this.savePrintEventDetails(requestIds);
+    }
+    return new Promise(resolve => {
       this.context.sendCallout({ message: <FormattedMessage id="ui-requests.printInProgress" /> });
       onToggle();
       // without the timeout the printing process starts right away
       // and the callout and onToggle above are blocked
       setTimeout(() => resolve(), 1000);
-    })
-  );
+    });
+  };
 
-  onBeforeGetContentForSinglePrintButton = () => (
-    new Promise(resolve => {
+  onBeforeGetContentForSinglePrintButton = (requestId) => {
+    this.savePrintEventDetails([requestId]); // add enable print log settings check
```

Comment:
this will trigger the savePrintEventDetails once user click on 'Print' or 'Cancel' button in PDF Preview as "onAfterPrint"gets called after button click in PDF Preview. That is not mentioned in the acceptance criteria. I will check with team on this.

Reply:
onAfterPrint is not used here as we do not have control over react-to-print library and we need to save logs once user clicks any of Print button on Request App. I did shifted **onBeforeGetContent** to **onBeforePrint**.